metricflow/engine/metricflow_engine.py

```diff
@@ -712,22 +712,10 @@ def get_dimension_values(  # noqa: D102
                 query_type=MetricFlowQueryType.DIMENSION_VALUES,
             )
         )
-        result_dataframe = query_result.result_df
-        if result_dataframe is None:
+        if query_result.result_df is None:
             return []
 
-        # Snowflake likes upper-casing things in result output, so we lower-case all names
-        # before operating on the dataframe.
-        metric_names = [metric_name.lower() for metric_name in metric_names]
-        result_dataframe.columns = result_dataframe.columns.str.lower()
-
-        # Get dimension values regardless of input name -> output dimension mapping. This is necessary befcause
-        # granularity adjustments on time dimensions produce different output names for dimension values.
-        # Note: this only works as long as we have exactly one column of group by values
-        # and no other extraneous output columns
-        dim_vals = result_dataframe[result_dataframe.columns[~result_dataframe.columns.isin(metric_names)]].iloc[:, 0]
-
-        return sorted([str(val) for val in dim_vals])
+        return sorted([str(val) for val in query_result.result_df.column_values_iterator(0)])
```

Comment:
Yeah, there's a specified order when the SQL is rendered to have the dimension values first

https://github.com/dbt-labs/metricflow/blob/bbd2901522e3b8926de4e733571af13d88eba606/metricflow/plan_conversion/select_column_gen.py#L36

It would be better to do a lookup, but mapping the name to the output column is not as straightforward as it should be since the output column name can be different from the input.